### PR TITLE
Check that GUIDs are unique

### PR DESCRIPTION
**Pull request by iwsmith**
Check configs to make sure GUIDs used by Slates and SlateLineups are unique.

## Update
Based on PR feedback I did the following:
* Moved slate and lineup validation code into their respective models. This means we do the validations on model load, which is probably right so we don't end up with invalid models being loaded
* Cross GUID checks (between slate and lineup) done at app start.
* Tests

### CODE REVIEW

File: tests/functional/json/slate_lineup_config.py [hunk 1]
```diff
@@ -1,15 +1,32 @@
...
+            raise ValueError(f"Slate GUID {sid} appears more than once in slate config")
+
+    lineup_ids = Counter([r.id for r in SlateLineupConfigModel.SLATE_LINEUP_CONFIGS])
+    for lid, count in lineup_ids.items():
+        if count > 1:
+            raise ValueError(f"SlateLineup GUID {lid} appears more than once in slate lineup config")
+
+    guid_overlap = set(lineup_ids.keys()).intersection(slate_ids.keys())
+    if guid_overlap:
+        raise ValueError(f"GUID is reused for slate and lineup: {guid_overlap}")
```
Comment (chelseatroy) [inline tests/functional/json/slate_lineup_config.py:27]: There are faster ways to do this, but they either require an extra step to get the actual duplicated ID or aren't as clear to follow as this, so I'm happy with this solution.

**Review by chelseatroy — CHANGES_REQUESTED** [review]
The solution I'm happy with. I think it needs three automated tests:

- one for duped slate ids
- one for duped layout ids
- one for reused slate ids among the layout ids

**Review by jpetto — APPROVED** [review]
looks solid to me 🙌

**Review by kkyeboah — COMMENTED** [review]
This works for me but deferring to @chelseatroy for the requested changes.

**Comment by iwsmith** [comment]
> The solution I'm happy with. I think it needs three automated tests:
> 
>     * one for duped slate ids
> 
>     * one for duped layout ids
> 
>     * one for reused slate ids among the layout ids
> 
> 
> Happy to pair on that if that's helpful!

Hmm, I've never written a test for a test before. I think this really indicates this code shouldn't be in a unit test, but instead somewhere else. I will move stuff around and resubmit.

**Review by chelseatroy — APPROVED** [review]
That's the stuff!

**Pull request merged**